providers/telekom.go

```diff
@@ -0,0 +1,155 @@
+// Telekom Cloud OTC DNS management for the BBB autoscaler implementation
+// Config of Telekom Cloud
+// Creation of a new OTC DNS record
+// Deletion of a existing OTC DNS record
+// Updating of the IP in a OTC DNS record
+package providers
```

Comment:
The structure isn't right, Telekom doesn't implement the Provider interface, so it shouldn't be in the providers package, we should create a separate interface and package for DNS providers.